src/solvers/dg/dg_2d.jl

```diff
@@ -24,6 +24,12 @@ function create_cache(mesh::TreeMesh{2}, equations::AbstractEquations{2},
   cache = (;cache..., create_cache(mesh, equations, dg.volume_integral, dg, uEltype)...)
   cache = (;cache..., create_cache(mesh, equations, dg.mortar, uEltype)...)
 
+  du_ec = Array{uEltype, 4}(undef, nvariables(equations), nnodes(dg), nnodes(dg),
+                            nelements(dg, cache))
+  du_cen = Array{uEltype, 4}(undef, nvariables(equations), nnodes(dg), nnodes(dg),
+                             nelements(dg, cache))
+  cache = (;cache..., du_ec, du_cen)
```

Comment:
I would prefer to move this to an appropriate call `cache = (;cache..., create_cache(mesh, equations, dg.volume_integral, dg, uEltype)...)` (with a new volume integral type) and allocate less memory by requiring only `Threads.nthreads()` times the space for one element. But that would require changing the split form kernel, too, so we can also keep the larger allocations for now.

Reply:
Agreed! Let's keep this on the TODO list.